plugins/inputs/logstash/logstash.go

```
@@ -0,0 +1,290 @@
+	}
+
+	now := time.Now()
+	// Events
+	for p, s := range stats {
+		f := jsonparser.JSONFlattener{}
+		// parse Json, ignoring strings and bools
+		err := f.FlattenJSON("", s)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("logstash_"+p, f.Fields, tags, now)
+	}
+
+	// Input Plugins
+	for _, plugin := range PipelineStats.Pipeline.Plugins.Inputs {
+		//plugin := &plugin
+		fields := map[string]interface{}{
+			"name":               plugin.Name,
+			"duration_in_millis": plugin.Events.DurationInMillis,
+			"in":                 plugin.Events.In,
+			"out":                plugin.Events.Out,
+		}
+		acc.AddFields("logstash_plugin_input_"+plugin.Name, fields, tags, now)
+	}
+
+	// Filters Plugins
+	for _, plugin := range PipelineStats.Pipeline.Plugins.Filters {
+		//plugin := &plugin
+		fields := map[string]interface{}{
+			"name":               plugin.Name,
+			"duration_in_millis": plugin.Events.DurationInMillis,
+			"in":                 plugin.Events.In,
+			"out":                plugin.Events.Out,
+		}
+		acc.AddFields("logstash_plugin_filter_"+plugin.Name, fields, tags, now)
+	}
+
+	// Output Plugins
+	for _, plugin := range PipelineStats.Pipeline.Plugins.Outputs {
+		//plugin := &plugin
+		fields := map[string]interface{}{
+			"name":               plugin.Name,
+			"duration_in_millis": plugin.Events.DurationInMillis,
+			"in":                 plugin.Events.In,
+			"out":                plugin.Events.Out,
+		}
+		acc.AddFields("logstash_plugin_output_"+plugin.Name, fields, tags, now)
+	}
+
+	return nil
+}
+
+//Gather is main function to gather all metrics provided by this plugin
+func (l *Logstash) Gather(acc telegraf.Accumulator) error {
+
+	if l.client == nil {
+		client, err := l.createHTTPClient()
+
+		if err != nil {
+			return err
+		}
+		l.client = client
+	}
+
+	if err := l.gatherJVMStats(l.LogstashURL+jvmStats, acc); err != nil {
+		err = fmt.Errorf("Can't gather JVM Stats from: " + l.LogstashURL + jvmStats)
```

It would be better to return the error unmodified so that we don't hide the cause.